src/test/java/com/softserve/academy/spaced/repetition/service/AccessToUrlServiceTest.java

```diff
@@ -0,0 +1,283 @@
+
+import static org.junit.Assert.assertEquals;
+import static org.mockito.Mockito.*;
+
+@RunWith(MockitoJUnitRunner.class)
+@Transactional
+public class AccessToUrlServiceTest {
+
+    private final Long CATEGORY_ID = 1L;
+    private final Long COURSE_ID = 1L;
+    private final Long CARD_ID = 1L;
+    private final Long FOLDER_ID = 1L;
+    private final Long DECK_ID = 1L;
+    private final Long COMMENT_ID = 1L;
+    private Category category;
+    private List<Course> courseList;
+    private List<Deck> deckList;
+
+    @Mock
+    private CategoryRepository categoryRepository;
+
+    @Mock
+    private UserService userService;
+
+    @Mock
+    private CourseRepository courseRepository;
+
+    @Mock
+    private FolderRepository folderRepository;
+
+    @Mock
+    private DeckRepository deckRepository;
+
+    @Mock
+    private CardRepository cardRepository;
+
+    @Mock
+    private DeckCommentRepository deckCommentRepository;
+
+    @Mock
+    private CourseCommentRepository courseCommentRepository;
+
+    @InjectMocks
+    private AccessToUrlServiceImpl accessToUrlService;
+
+    @Before
+    public void setUp() throws Exception {
+        courseList = new ArrayList<>();
+        deckList = new ArrayList<>();
+        List<Card> cardList = new ArrayList<>();
+        User user = DomainFactory.createUser(1L, new Account(), new Person(), new Folder(), null);
+        Course course = DomainFactory.createCourse(COURSE_ID, null, null, null, 1L,
+                true, null, category, deckList, null, null);
+        category = DomainFactory.createCategory(CATEGORY_ID, null, null,
+                null, null, null);
+        Deck deck = DomainFactory.createDeck(DECK_ID, null, null, null, category
+                , 1L, user, cardList, courseList, null, null, null);
+        Card card = DomainFactory.createCard(CARD_ID, null, null, null, deck);
+        cardList.add(card);
+        courseList.add(course);
+        deckList.add(deck);
+        user.getAccount().setAuthorities(createAuthoritySet());
+        when(userService.getAuthorizedUser()).thenReturn(user);
+        when(courseRepository.getAccessToCourse(CATEGORY_ID, COURSE_ID))
+                .thenReturn(courseList);
+        when(deckRepository.hasAccessToDeck(COURSE_ID, DECK_ID))
+                .thenReturn(deckList);
+        when(cardRepository.hasAccessToCard(DECK_ID, CARD_ID))
+                .thenReturn(cardList);
+    }
+
+    private Set<Authority> createAuthoritySet() {
+        Set<Authority> authorities = new HashSet<>();
+        Authority roleAdmin = new Authority();
+        Authority roleUser = new Authority();
+        roleAdmin.setName(AuthorityName.ROLE_ADMIN);
+        roleUser.setName(AuthorityName.ROLE_USER);
+        authorities.add(roleAdmin);
+        authorities.add(roleUser);
+        return authorities;
+    }
+
+    @Test
+    public void testHasAccessToCategory() {
+        List<Category> categoryList = new ArrayList<>();
+        categoryList.add(category);
```

Comment:
No need to add some elements, because you can compare on false
And, if i'm not mistaken, we have method like `assertFalse` or smt like that, but i'n not sure)

Reply:
Some methods works with elements of list, so the resullt of test with empty list may not be actual.

Reply:
If this list uses in other methods, why do you create it here, not in `setUp` method?